.github/workflows/tests.yml

```diff
@@ -3,26 +3,25 @@ on: [push, pull_request]
 name: Tests
 jobs:
   test:
-    name: Test suite
+    strategy:
+      matrix:
+        net: [main, test, beta]
+    name: Test suite (${{ matrix.net }}net)
     runs-on: self-hosted
     steps:
       - name: Clone the repository
         uses: actions/checkout@v2
       - run: echo "$HOME/.cargo/bin" >> $GITHUB_PATH
       - name: Prepare target directory
+        run: rm -rf target && cp -r ~/build-cache/aurora-engine target
+      - run: make ${{ matrix.net }}net-test-build
+      - name: Run ${{ matrix.net }}net cargo test
+        run: cargo test --locked --verbose --features ${{ matrix.net }}net-test
+      - name: Cache and clear target directory
         run: |
-          rm -rf target && mkdir target
-          ln -s ~/build-cache/aurora-engine/debug/ target/debug
-          ln -s ~/build-cache/aurora-engine/release/ target/release
-      - run: make mainnet-test-build
-      - run: make testnet-test-build
-      - run: make betanet-test-build
-      - name: Run mainnet cargo test
-        run: cargo test --locked --verbose --features mainnet-test
-      - name: Run testnet cargo test
-        run: cargo test --locked --verbose --features testnet-test
-      - name: Run betanet cargo test
-        run: cargo test --locked --verbose --features betanet-test
+          mv -n target/debug/* ~/build-cache/aurora-engine/debug
+          mv -n target/release/* ~/build-cache/aurora-engine/release
```

Comment:
I don't think `no-clobber` will be good enough because we may want to overwrite some files that are updated sometimes. To be honest I'm not 100% sure on this because I don't know exactly how rust compilation artifacts are represented. Regardless, I think `rsync` might be a better choice here.

I also think it would be good to update the whole `target` directory, not just `debug` and `release` because there are some other build artifacts we care about in `target` not under those directories (e.g. solidity build artifacts and wasm build artifacts).

Reply:
I've came to the exactly same conclusion.
Artifacts in `debug` in `release` are suffixed with some hash, so I assumed that compiler kinda only creates artifacts with new hash instead of overwriting existing ones. But it seems to me that it's not true, because cleaning build-cache and starting from scratch gives the same time result :(

So this new solution doesn't help at all, you are right. I'll delete it.
I'm now thinking about implementing the same thing that we had in `actions-rs/cache`, but saving cache locally, and without using compression. I'll do this in another commit. Thanks for details!